saleor/tests/e2e/checkout/test_checkout_calculate_correct_discount_for_sale_and_voucher.py

```diff
@@ -0,0 +1,181 @@
+import pytest
+
+from ..product.utils.preparing_product import prepare_product
+from ..sales.utils import create_sale, create_sale_channel_listing, sale_catalogues_add
+from ..shop.utils.preparing_shop import prepare_shop
+from ..utils import assign_permissions
+from ..vouchers.utils import create_voucher, create_voucher_channel_listing
+from .utils import (
+    checkout_add_promo_code,
+    checkout_complete,
+    checkout_create,
+    checkout_delivery_method_update,
+    checkout_dummy_payment_create,
+    checkout_lines_add,
+)
+
+
+def prepare_sale_for_variant(e2e_staff_api_client, channel_id, product_variant_id):
+    sale_name = "Sale PERCENTAGE"
+    sale_discount_type = "PERCENTAGE"
+    sale_discount_value = 13
+    sale = create_sale(e2e_staff_api_client, sale_name, sale_discount_type)
+    sale_id = sale["id"]
+    sale_listing_input = [
+        {
+            "channelId": channel_id,
+            "discountValue": sale_discount_value,
+        }
+    ]
+    create_sale_channel_listing(
+        e2e_staff_api_client, sale_id, add_channels=sale_listing_input
+    )
+    catalogue_input = {"variants": [product_variant_id]}
+    sale_catalogues_add(e2e_staff_api_client, sale_id, catalogue_input)
+
+    return sale_id, sale_discount_value
+
+
+def prepare_voucher(
+    e2e_staff_api_client, channel_id, voucher_code, voucher_discount_value
+):
+    voucher_code = "VOUCHER001"
+    voucher_data = create_voucher(
+        e2e_staff_api_client,
+        "PERCENTAGE",
+        voucher_code,
+        "ENTIRE_ORDER",
+    )
+    voucher_id = voucher_data["id"]
+
+    channel_listing = [
+        {"channelId": channel_id, "discountValue": voucher_discount_value}
+    ]
+    create_voucher_channel_listing(e2e_staff_api_client, voucher_id, channel_listing)
+
+    return voucher_discount_value, voucher_code
+
+
+@pytest.mark.e2e
+def test_checkout_calculate_discount_for_sale_and_voucher_1014(
```

Comment:
Wouldn't be good to have tests for scenarios with different sale and voucher type?
Like:
sale: fixed, voucher: percentage
sale: percentage, voucher:fixed
sale:fixed, voucher:fixed

Reply:
I added parameterization